examples/openui5/js/control/ToDoControl.js

```diff
@@ -0,0 +1,348 @@
+
+			// destroy toDo if new title is empty
+			if (!newTitle || +newTitle.length === 0) {
+				this.fireEvent('deleteToDoPressed', {
+					toDoId: +toDoId
+				});
+			} else {
+				this.fireEvent('toDoChangedPressed', {
+					toDoId: +toDoId,
+					title: newTitle
+				});
+			}
+
+			this.rerender();
+		},
+
+		onToggleAllToDos: function () {
+			var toDoIds = this._getAllToDosWithStatus(false);
+			var elems = this.getToDosObject();
+
+			if (toDoIds.length > 0) {
+				// complete all uncompleted toDos
+				toDoIds.forEach(function (value) {
+					// fire changed event
+					this.fireEvent('completedToDoPressed', {
+						toDoId: value,
+						completed: true
+					});
+				}, this);
+
+			} else {
+				// uncomplete all toDos
+				elems.forEach(function (value) {
+					// fire changed event
+					this.fireEvent('completedToDoPressed', {
+						toDoId: value.id,
+						completed: false
+					});
+				}, this);
+			}
+
+			this.rerender();
+		},
+
+		/*
+		 * Set event bindings for click, double-click, focus events
+		 * use jQuery eventhandling since it's using addEventListener
+		 */
+		onAfterRendering: function () {
+			var controlId = this.getId();
+			var allToDosObjects = this.getToDosObject();
+
+			// what happens when user hits enter on text field?
+			$('#inputToDo-' + controlId).keypress(this.onAddToDo.bind(this));
+
+			// what happens when user clicks on the down arrow left to the input field?
+			$('#inputToDoToggle-' + controlId).click(this.onToggleAllToDos.bind(this));
+
+			// what happens when user clicks on 'Clear completed' link in the bottom right?
+			$('#deleteAllCompleted-' + controlId).click(this.onDeleteAllCompletedToDos.bind(this));
+
+			allToDosObjects.forEach(function (value) {
+				// what happens when user delets toDo?
+				$('#destroyToDo-' + value.id + '-' + controlId).click(this.onDeleteToDo.bind(this));
+
+				// what happens when user double-clicks a toDo?
+				$('#labelFor-' + value.id + '-' + controlId).dblclick(this.onEditToDo.bind(this));
+
+				// what happens when user removes focus while editing a toDo?
+				// captures blur events in IE as well
+				$('#editField-' + value.id + '-' + controlId).focusout(this.onLeaveEditToDoFocus.bind(this));
+
+				// what happens when user presses a key while editing a toDo?
+				$('#editField-' + value.id + '-' + controlId).keydown(this._onEditKeyPress.bind(this));
+
+				// what happens when user checks the completed checkbox for a toDo?
+				$('#completeToDo-' + value.id + '-' + controlId).click(this.onToggleToDo.bind(this));
+			}, this);
+
+			// set focus on input textfield
+			document.getElementById('inputToDo-' + this.sId).focus();
+		},
+
+		/*
+		 * Define how the ToDoMVC control should be rendered
+		 */
+		renderer: {
+			controlID,
+
+			render: function (oRm, oControl) {
+				controlID = oControl.getId();
+				toDos = oControl.getToDosObject();
+				filter = oControl.getFilter();
+				toDosAvailable = (toDos && toDos.length > 0) ? true : false;
```

Comment:
I'm curious, why the `? true : false` here? `toDos && toDos.length > 0` is a boolean anyhow right?